### ISSUE TITLE
Spell/Script: Thrust remove two shield layers instead of one

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#22193]
**Description:** The spell https://woehead.way-of-elendil.fr/?spell=62544 remove 2 shield layers instead of one.

**Steps to reproduce the problem:**

1. Connect an alliance char that doesn't starts the argent tournament.
2. .Quest add  13667   and .quest rew  13667
3. .quest add 13828  and .add 46069
4. .go c 75473 equip the spare and ride the horse
5. activate 3 shields
6. go to a melee target and use the spell Thrust: https://woehead.way-of-elendil.fr/?spell=62544
7. Notice 2 shield layers are removed instead of one

related script: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Spells/spell_generic.cpp#L660-L666

**Branch(es):** 3.3.5

TrinityCore rev. b7c0f1961c88 2018-07-24 10:46:14 +0200 (3.3.5 branch) (Win64, Release, Static)
Using SSL version: OpenSSL 1.0.2o  27 Mar 2018 (library: OpenSSL 1.0.2o  27 Mar 2018)
Using Boost version: 1.67.0
Using MySQL version: 5.6.15
Using CMake version: 3.11.2
Compiled on: Windows 
Automatic database updates are disabled for all databases!
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in ./maps. Total size: 252191207 bytes
vmaps directory located in ./vmaps. Total size: 588247501 bytes
mmaps directory located in ./mmaps. Total size: 2229000480 bytes
Using frFR DBC Locale as default. All available DBC locales: frFR 
Using World DB: TDB 335.64

**Comment by illfated** [comment]
Spell https://woehead.way-of-elendil.fr/?spell=62544#used-by-npc only lists NPCs as users of this spell.

Is it only due to missing information on that page, meaning that players use that spell too, even if not listed as such?

http://www.wowhead.com/spell=62544/thrust does not list any NPCs using the spell, it is listed under http://www.wowhead.com/uncategorized-spells -- is this more "correct" than the https://woehead.way-of-elendil.fr/?spell=62544 page?

**Comment by Killyana** [comment]
No players use the spell via vehicles (npcs) ,the spell will appear on the used-by-npc if it's used on the SAI or set on spell1,2,3 ect... on creature_template

**Comment by nagrandbugtracker** [comment]
Confirmed on latest revision.

**Comment by illfated** [comment]
@nagrandbugtracker : Please provide TC revision hash / commit + which platform/OS used to reproduce this issue (edit your post if you can).
"on latest revision." is not practical for other users to look up, since it requires checking the exact date & time (& timezone) of your comment before searching the TC repository.

TL;DR: please use the actual commit hash showing up as a link in your posts when confirming. Server OS info is useful too.

**Comment by PolarCookie** [comment]
Tested again, it's fixed now.
You can close this.
